src/inject/dynamic-theme/index.ts

```diff
@@ -252,16 +258,54 @@ function createManager(element: StyleElement) {
     return manager;
 }
 
-function updateVariables(newVars: Map<string, string>) {
+function updateVariables(newVars: Map<string, DarkReaderVariable>) {
     if (newVars.size === 0) {
         return;
     }
     newVars.forEach((value, key) => {
+        delete parsedVariables[`${key};${value.property}`];
         variables.set(key, value);
     });
+    const variablesStyle: HTMLStyleElement = createOrUpdateStyle('darkreader--dynamicVariable');
+    const {sheet} = variablesStyle;
+
+    for (let i = sheet.cssRules.length - 1; i >= 0; i--) {
+        sheet.deleteRule(i);
+    }
+
     variables.forEach((value, key) => {
-        variables.set(key, replaceCSSVariables(value, variables));
+        if (parsedVariables[`${key};${value.property}`]) {
+            const {modifiedBackground, modifiedText} = parsedVariables[`${key};${value.property}`];
+            sheet.insertRule([
+                `${key} {`,
+                `   --darkreader-bg${value.property}: ${modifiedBackground};`,
+                `   --darkreader-text${value.property}: ${modifiedText};`,
+                `}`
+            ].join('\n'));
+        } else {
+            let parsedValue: RGBA;
+            try {
+                parsedValue = parse(value.value);
+            } catch (err) {
+                logWarn(err);
```

Comment:
Only fails for nested variables.
e.g.

```CSS
    --color-red: #ccc;
    --proxy-color-red: var(--color-red);
```
Will throw an error as `var(--color-red)` should be replaced with the value of Dark Reader's modified one however. It doesn't have to be in order so.

```CSS
    --proxy-color-red: var(--color-red);
    --color-red: #ccc;
```
And Dark Reader won't have a single knowledge about --color-red *yet* and should be placed on some list.